components/automate-ui/src/app/services/projects-filter/projects-filter.reducer.ts

```diff
@@ -78,13 +78,16 @@ export function projectsFilterReducer(
     }
 
     case ProjectsFilterActionTypes.SAVE_OPTIONS: {
+      // merge any added or removed options with the newly saved values
+      const mergedOptions = mergeOptions(state.options, action.payload);
+      const sortedOptions = sortOptions(mergedOptions);
       return pipe(
-        set('options', action.payload),
-        set('selectionLabel', selectionLabel(action.payload)),
-        set('selectionCount', selectionCount(action.payload)),
-        set('selectionCountVisible', selectionCountVisible(action.payload)),
-        set('selectionCountActive', selectionCountActive(action.payload)),
-        set('dropdownCaretVisible', dropdownCaretVisible(action.payload))
+        set('options', sortedOptions),
+        set('selectionLabel', selectionLabel(sortedOptions)),
+        set('selectionCount', selectionCount(sortedOptions)),
+        set('selectionCountVisible', selectionCountVisible(sortedOptions)),
+        set('selectionCountActive', selectionCountActive(sortedOptions)),
+        set('dropdownCaretVisible', dropdownCaretVisible(sortedOptions))
```

Comment:
Since this block is now virtually identical to that for `LOAD_OPTIONS_SUCCESS`, seems like it would ease the future maintenance burden by extracting a common `setState` function? There are just two differences to account for -- `filterVisible` and `optionsLoadingStatus` -- and I believe it would essentially be a no-op if those were added here for `SAVE_OPTIONS`, too.
Then, this block would just become
```
case ProjectsFilterActionTypes.SAVE_OPTIONS: { 
  return setState(state.options, action.payload);
}
```

Reply:
I started to take a shot at this but I think it actually makes things more complex because LOAD_OPTIONS_SUCCESS has two additional fields to set and one of them wants the sorted options:

```
        set('optionsLoadingStatus', EntityStatus.loadingSuccess),
        set('filterVisible', filterVisible(sortedOptions))
```